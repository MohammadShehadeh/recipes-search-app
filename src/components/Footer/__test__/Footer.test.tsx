import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';

import { Footer } from '..';

jest.useFakeTimers().setSystemTime(new Date('2023-12-10'));

describe('Footer', () => {
  let dateNowSpy: any;

  beforeAll(() => {
    dateNowSpy = jest.spyOn(Date, 'now');
  });

  afterAll(() => {
    dateNowSpy.mockRestore();
  });

  it('should render to match snapshot', () => {
    const tree = renderer.create(<Footer />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it('should render Footer correctly', () => {
    render(<Footer />);
    const footerElement = screen.getByText(/Copyright Â© 2023 Mohammad Shehadeh. All rights reserved/i);
    expect(footerElement).toBeInTheDocument();
  });
});
