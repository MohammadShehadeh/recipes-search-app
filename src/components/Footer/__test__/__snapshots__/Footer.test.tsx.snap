// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer should render to match snapshot 1`] = `
<footer
  className="footer"
>
  <div
    className="container"
  >
    <div
      className="main"
    >
      Copyright Â© 
      2023
       Mohammad Shehadeh. All rights reserved.
    </div>
  </div>
</footer>
`;
